---
title: "Mutational Analysis"
output: html_document
date: "2025-11-20"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(ggplot2)
library(dplyr)
library(survival)
library(survminer)
```


#The first thing we want to do before we analyze all of the data is to check and isolate the patients which have all three datasets which is clinical data, mutation data, and expression data. 

```{r}

#First we will read all three data sets 

clinical_data <- read.delim("data_clinical_patient.txt", header = TRUE, stringsAsFactors = FALSE)
mutation_data <- read.delim("data_mutations.txt", header = TRUE, stringsAsFactors = FALSE)
rnaseq_data <- read.csv("RNAseq_LIHC.csv", stringsAsFactors = FALSE)
```


```{r}
extract_id <- function(tcga_codes) {
  tcga_codes <- as.character(tcga_codes)

  ids <- sub("^TCGA-[A-Z0-9]{2}-([A-Z0-9]{4}).*", "\\1", tcga_codes)
  
  return(ids)
}
```


```{r}
extract_id_rnaseq <- function(tcga_codes) {
  tcga_codes <- as.character(tcga_codes)

  ids <- sub("^TCGA.[A-Z0-9]{2}.([A-Z0-9]{4}).*", "\\1", tcga_codes)
  
  return(ids)
}
```


```{r}
clinical_data_patients_id <- extract_id(tcga_codes = clinical_data$X.Patient.Identifier)
clinical_data_patients_id <- clinical_data_patients_id[-(1:4)]

mutational_data_patients_id <- unique(extract_id(tcga_codes = mutation_data$Tumor_Sample_Barcode))

rnaseq_data_cleaned <- rnaseq_data[,-1 ]

rnaseq_data_patients_id <- unique(extract_id_rnaseq(tcga_codes = colnames(rnaseq_data)))

common_patients_id <- clinical_data_patients_id[clinical_data_patients_id %in% mutational_data_patients_id &
clinical_data_patients_id %in% rnaseq_data_patients_id ]
```


```{r}
clinical_full_data <- clinical_data[ which(clinical_data_patients_id  %in% common_patients_id),  ]


```


```{r}
colnames(mutation_data)
```
```{r}


build_gene_patient_matrix <- function(mut_df,
                                      variant_mode = c("all", "non_synonymous", "synonymous")) {

  variant_mode <- match.arg(variant_mode)

  # Add patient_id extracted from Tumor_Sample_Barcode
  mut_df$patient_id <- extract_id(mut_df$Tumor_Sample_Barcode)

  # Define mutation type groups
  nonsynonymous_classes <- c(
    "Missense_Mutation",
    "Nonsense_Mutation",
    "Frame_Shift_Del",
    "Frame_Shift_Ins",
    "In_Frame_Del",
    "In_Frame_Ins",
    "Splice_Site",
    "Translation_Start_Site",
    "Nonstop_Mutation",
    "Splice_Region"
  )

  synonymous_classes <- c("Silent")

  # Filter by variant type
  if (variant_mode == "non_synonymous") {
    mut_df <- mut_df[mut_df$Variant_Classification %in% nonsynonymous_classes, ]

  } else if (variant_mode == "synonymous") {
    mut_df <- mut_df[mut_df$Variant_Classification %in% synonymous_classes, ]
  }

  # Clean missing values
  mut_df <- mut_df[
    !is.na(mut_df$patient_id) & mut_df$patient_id != "" &
    !is.na(mut_df$Hugo_Symbol) & mut_df$Hugo_Symbol != "",
  ]

  # Build patient Ã— gene count matrix
  tab <- table(mut_df$patient_id, mut_df$Hugo_Symbol)

  # Convert to 0/1 matrix
  mat <- ifelse(as.matrix(tab) > 0, 1L, 0L)

  return(t(mat))
}


```


```{r}
matrix <- build_gene_patient_matrix(mutation_data)

gene_freq <- colSums(matrix)

# Sort genes by frequency (descending)
gene_freq_sorted <- sort(gene_freq, decreasing = TRUE)

# Pick top 20
top_n <- 20
top_genes <- names(gene_freq_sorted)[1:top_n]

top_genes


mat_top20 <- matrix[, top_genes, drop = FALSE]


```
```{r}

dist_mat <- dist(mat_top20, method = "euclidean")
hc_ward <- hclust(dist_mat, method = "ward.D2")
plot(hc_ward, labels = FALSE, main = "Hierarchical Clustering of Patients (Top 20 Mutated Genes)")
```
```{r}
clinical_data$Overall.Survival.Time <-  as.numeric(clinical_data$Overall.Survival.Status)

surv_obj <- Surv(time = clinical_data$Overall.Survival.Time,
                 event = clinical_data$Overall.Survival.Status)
```

