---
title: "uju's initial analysis"
output: html_document
date: "2025-11-21"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)


#bleh bleh bleh
```

```{r}
clinical <- read.delim("data_clinical_patient.txt", sep="\t", header=TRUE)
mutations <- read.delim("data_mutations.txt", sep="\t", header=TRUE)
rna <- read.csv("RNAseq_LIHC.csv", header=TRUE, row.names=1)

dim(clinical)
dim(mutations)
dim(rna)
head(rna[,1:5])

# 424 RNA-seq samples, 376 clinical records, 114 mutation profiles
```

```{r}
### code from lab 7
# before analysis we first filtered the data to only count patients in all three datasets
extract_id <- function(tcga_codes) {
  tcga_codes <- as.character(tcga_codes)
  ids <- sub("^TCGA-[A-Z0-9]{2}-([A-Z0-9]{4}).*", "\\1", tcga_codes)
  return(ids)
}

extract_id_rnaseq <- function(tcga_codes) {
  tcga_codes <- as.character(tcga_codes)
  ids <- sub("^TCGA.[A-Z0-9]{2}.([A-Z0-9]{4}).*", "\\1", tcga_codes)
  return(ids)
}

#extract id
clinical_ids <- extract_id(clinical$X.Patient.Identifier)
clinical_ids <- clinical_ids[-(1:4)]   # removes first 4 header rows 

mutation_ids <- unique(extract_id(mutations$Tumor_Sample_Barcode))

rnaseq_ids <- unique(extract_id_rnaseq(colnames(rna)))

# finds the common ids. 
common_ids <- clinical_ids[
  clinical_ids %in% mutation_ids &
  clinical_ids %in% rnaseq_ids
]

cat("Number of patients present in ALL THREE datasets:", length(common_ids), "\n")
```


```{r}
clinical_filtered <- clinical[extract_id(clinical$X.Patient.Identifier) %in% common_ids, ]
mutation_filtered <- mutations[extract_id(mutations$Tumor_Sample_Barcode) %in% common_ids, ]
rnaseq_filtered <- rna[, extract_id_rnaseq(colnames(rna)) %in% common_ids]


cat("Filtered clinical:", nrow(clinical_filtered), "\n")
cat("Filtered mutation rows:", nrow(mutation_filtered), "\n")
cat("Filtered rnaseq columns:", ncol(rnaseq_filtered), "\n")
### guess some patients have more than 1 rna seq sample
```
```{r}
### mutation frequency analysis
library(dplyr)
library(ggplot2)

# to count mutations per gene 
gene_counts <- mutation_filtered %>%
  count(Hugo_Symbol, name = "Mutation_Count") %>%
  arrange(desc(Mutation_Count))

head(gene_counts, 20)


top20_genes <- gene_counts %>% slice_max(Mutation_Count, n = 20)

ggplot(top20_genes, aes(x = reorder(Hugo_Symbol, Mutation_Count), y = Mutation_Count)) +
  geom_bar(stat = "identity", fill = "steelblue") +
  coord_flip() +
  labs(title = "Top 20 Most Mutated Genes in LIHC",
       x = "Gene",
       y = "Number of Mutations") +
  theme_minimal()

### The most frequently mutated genes in our LIHC dataset were TTN, TP53, MUC16, and CTNNB1. 
#TTN and MUC16 are extremely large genes and are commonly mutated across many cancer types due to their size, making them typical high-frequency passenger mutations in TCGA datasets.

#In the main paper, TP53 and CTNNB1 were two of the most significantly mutated driver genes, with TP53 altered in 31% of tumors and CTNNB1 mutated in 27%. TP53 was classified as a recurrently inactivated tumor suppressor, while CTNNB1 was identified as the major oncogene driving Wnt/β-catenin pathway activation. 
#Our dataset similarly shows TP53 and CTNNB1 among the highest frequency drivers, aligning closely with the results reported in the main paper.

#Overall, these results are consistent with the main paper and support that our dataset captures liver cancer genomic features.
```
```{r}
mutation_filtered$Patient_ID <- substr(mutation_filtered$Tumor_Sample_Barcode, 1, 12)

patient_mut_counts <- mutation_filtered %>%
  count(Patient_ID, name = "Mutation_Burden")

head(patient_mut_counts)

ggplot(patient_mut_counts, aes(x = Mutation_Burden)) +
  geom_histogram(bins = 30, fill = "orange", color = "black") +
  labs(title = "Distribution of Mutation Burden Across Patients",
       x = "Number of Mutations",
       y = "Patient Count") +
  theme_minimal()


clinical_filtered$Patient_ID <- substr(clinical_filtered$X.Patient.Identifier, 1, 12)

merged_data <- merge(patient_mut_counts, clinical_filtered,
                     by = "Patient_ID", all.x = TRUE)

### the histrogram is very right skewed. maybe we should check if high burden links to lower survival rate or soemthing
```




```{r}
### survival analysis portion
colnames(merged_data)
colnames(merged_data)
table(merged_data$Overall.Survival.Status)
# 1 is died, 0 is alive
merged_data$OS_event <- ifelse(grepl("DECEASED", merged_data$Overall.Survival.Status), 1, 0)


# survival time in months 
merged_data$OS_time <- as.numeric(merged_data$Overall.Survival..Months.)
summary(merged_data$OS_time)
table(merged_data$OS_event)


median_burden <- median(merged_data$Mutation_Burden, na.rm = TRUE)

# split the patients to high and low mutation burden (# of mutations) by the median
merged_data$MutBurden_Group <- ifelse(
  merged_data$Mutation_Burden >= median_burden,
  "High mutation burden",
  "Low mutation burden"
)

table(merged_data$MutBurden_Group)

library(survival)

fit <- survfit(Surv(OS_time, OS_event) ~ MutBurden_Group, data = merged_data)

# plot
plot(fit, col = c("blue", "red"), lty = 1:2,
     xlab = "Time (months)", ylab = "Survival probability",
     main = "Overall survival by mutation burden group (LIHC)")
legend("bottomleft",
       legend = levels(factor(merged_data$MutBurden_Group)),
       col = c("blue", "red"),
       lty = 1:2)


survdiff(Surv(OS_time, OS_event) ~ MutBurden_Group, data = merged_data)

sd <- survdiff(Surv(OS_time, OS_event) ~ MutBurden_Group, data = merged_data)
pval <- 1 - pchisq(sd$chisq, length(sd$n) - 1)
pval

### the high mutation burden group has worse survival since it drops faster 
# but the log rank test shows the results are borderline nonsignificant...








```
```{r}
# mutation burden vs tumor stage

stage_data <- merged_data %>%
  mutate(Stage_raw = Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code) %>%
  filter(!is.na(Stage_raw), Stage_raw != "") %>%
  mutate(
    Stage_simple = case_when(
      grepl("^STAGE I([^I]|$)", Stage_raw)   ~ "Stage I",   
      grepl("^STAGE II([^I]|$)", Stage_raw)  ~ "Stage II",  
      grepl("^STAGE III([^I]|$)", Stage_raw) ~ "Stage III", 
      grepl("^STAGE IV([^I]|$)", Stage_raw)  ~ "Stage IV",  
      TRUE ~ NA_character_        # need to do it like this so it doesnt only output stage I
    )
  ) %>%
  filter(!is.na(Stage_simple))

table(stage_data$Stage_raw)
table(stage_data$Stage_simple)


ggplot(stage_data, aes(x = Stage_simple, y = Mutation_Burden)) +
  geom_boxplot(fill = "lightblue") +
  labs(
    title = "Mutation burden across tumour stages (LIHC)",
    x = "Tumour stage",
    y = "Mutation burden"
  ) +
  theme_minimal()

kruskal_test_result <- kruskal.test(Mutation_Burden ~ Stage_simple, data = stage_data)
kruskal_test_result

stage_medians <- stage_data %>%
  group_by(Stage_simple) %>%
  summarise(
    n = n(),
    median_burden = median(Mutation_Burden),
    mean_burden = mean(Mutation_Burden)
  )
stage_medians

table(stage_data$Stage_simple)
kruskal_test_result

### A Kruskal–Wallis test confirmed a statistically significant association between tumour stage and mutation burden (χ² = 6.82, p = 0.033). This suggests that mutation burden may increase with disease progression, or that later-stage tumours exhibit greater genomic instability. The trend aligns with the biological expectation that more advanced tumours accumulate additional mutations over time.
# not sure why we dont use anova or something else but this kruskal-wallis (suggested by chat) does show statistical significance. 
```



______after reading the main paper______

```{r}
#these are the genes in the main paper 
key_genes <- c(
  "TP53", "CTNNB1", "AXIN1", "RB1",          # classic drivers
  "ALB", "APOB", "CPS1",                     # metabolic genes
  "TERT",                                    # promoter mutations if present
  "LZTR1", "EEF1A1", "SF3B1", "SMARCA4"      # “new” SMGs in their paper
)

# make sure Patient_ID exists
mutation_filtered$Patient_ID <- substr(mutation_filtered$Tumor_Sample_Barcode, 1, 12)

# count how many *patients* have a mutation in each key gene
driver_counts <- mutation_filtered %>%
  filter(Hugo_Symbol %in% key_genes) %>%
  distinct(Hugo_Symbol, Patient_ID) %>%        # one row per gene–patient pair
  count(Hugo_Symbol, name = "n_patients")

# total number of patients (in mutation data)
n_patients_mut <- length(unique(mutation_filtered$Patient_ID))

driver_counts <- driver_counts %>%
  mutate(freq = n_patients / n_patients_mut)

driver_counts

ggplot(driver_counts, aes(x = reorder(Hugo_Symbol, freq), y = freq)) +
  geom_bar(stat = "identity", fill = "red") +
  coord_flip() +
  scale_y_continuous(labels = scales::percent) +
  labs(
    title = "Mutation frequency of key LIHC driver genes",
    x = "Gene",
    y = "Patients with ≥1 mutation (%)"
  ) +
  theme_minimal()

## so we counted the frequecy of the key genes listed in the LIHC main paper.
# interesting that the frequecy of tp53 is 31 percent. excatly matches the data in the main paper
 

```



```{r}
##some more clincal data analysis
#colnames(clinical_filtered)

# Survival analysis by age

# Convert age column to numeric
merged_data$Age <- as.numeric(merged_data$Diagnosis.Age)
summary(merged_data$Age)

# Remove missing age values (if any)
age_data <- merged_data %>% filter(!is.na(Age))

# Median age split
median_age <- median(age_data$Age, na.rm = TRUE)
median_age

age_data$Age_Group <- ifelse(age_data$Age >= median_age,
                             "Older",
                             "Younger")

table(age_data$Age_Group)

# Kaplan-Meier survival analysis
fit_age <- survfit(Surv(OS_time, OS_event) ~ Age_Group, data = age_data)

plot(fit_age, col=c("blue","red"), lty=1:2,
     main="Overall Survival by Age Group (LIHC)",
     xlab="Time (months)",
     ylab="Survival Probability")

legend("bottomleft",
       legend=levels(factor(age_data$Age_Group)),
       col=c("blue","red"), lty=1:2)

# Log-rank test
age_test <- survdiff(Surv(OS_time, OS_event) ~ Age_Group, data = age_data)
age_pval <- 1 - pchisq(age_test$chisq, length(age_test$n) - 1)
age_pval

#Median age = 61, so we split patients into:
#Younger: <61 years (n = 172)
# Older: ≥61 years (n = 180)
#The older group has a lower survival probability, especially after ~50 months.But the difference is not statistically significant.
#The trend still matches clinical expectations: older LIHC patients tend to do worse. But lacks statical significance...
```
```{r}
#3rd try to find stat significant analysis
# analysis on tumor stage
stage_surv <- survfit(Surv(OS_time, OS_event) ~ Stage_simple, data = stage_data)

plot(stage_surv, col=1:4, lty=1,
     main="Survival by Tumor Stage (LIHC)")

legend("bottomleft", legend=levels(factor(stage_data$Stage_simple)),
       col=1:4, lty=1)

survdiff(Surv(OS_time, OS_event) ~ Stage_simple, data=stage_data)

### test to see wether ajcc tumor stage is associated with survival
#Stage I (n = 172), Stage II (n = 80), Stage III (n = 82)
#Log-rank test:χ² = 18.1, df = 2, p = 0.0001 --> meaning the stage-related survival differences are statistically significant.
# also, this trend is clinically to be expected as later-stage LIHC generally has worse outcomes

```


```{r}
#some more clincal data anylsis
#colnames(clinical_filtered)
stage_data <- merged_data %>%
  mutate(Stage_raw = Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code) %>%
  filter(!is.na(Stage_raw), Stage_raw != "") %>%
  mutate(
    Stage_simple = case_when(
      grepl("^STAGE I([^I]|$)", Stage_raw)   ~ "Stage I",   
      grepl("^STAGE II([^I]|$)", Stage_raw)  ~ "Stage II",  
      grepl("^STAGE III([^I]|$)", Stage_raw) ~ "Stage III", 
      grepl("^STAGE IV([^I]|$)", Stage_raw)  ~ "Stage IV",  
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(Stage_simple))

# now add DSS time + event
stage_data$DSS_time <- suppressWarnings(
  as.numeric(stage_data$Months.of.disease.specific.survival)
)

stage_data$DSS_event <- ifelse(
  grepl("DECEASED", stage_data$Disease.specific.Survival.status),
  1,
  0
)

# keep only rows with valid DSS time
stage_dss <- stage_data %>% filter(!is.na(DSS_time))

library(survival)

fit_dss <- survfit(Surv(DSS_time, DSS_event) ~ Stage_simple, data = stage_dss)
plot(fit_dss,
     main = "Disease-specific survival by tumour stage",
     xlab = "Time (months)", ylab = "DSS probability")

survdiff(Surv(DSS_time, DSS_event) ~ Stage_simple, data = stage_dss)

#“Disease-specific survival (DSS) could not be analyzed because all patients in the filtered cohort had DSS_event = 0 (no recorded disease-specific deaths). This results in undefined statistics in the log-rank test. Therefore, DSS was excluded from downstream analyses. Overall survival (OS) was used as the primary clinical endpoint.” -chat
table(merged_data$DSS_event)
#so theres 0 desease specific deaths. so on to the next analysis. we cant use this one.
```


```{r}
table(merged_data$American.Joint.Committee.on.Cancer.Tumor.Stage.Code)

merged_data$T_stage <- merged_data$American.Joint.Committee.on.Cancer.Tumor.Stage.Code

# Simplify categories
merged_data$T_stage_simple <- case_when(
  grepl("^T1", merged_data$T_stage) ~ "T1",
  grepl("^T2", merged_data$T_stage) ~ "T2",
  grepl("^T3", merged_data$T_stage) ~ "T3",
  grepl("^T4", merged_data$T_stage) ~ "T4",
  TRUE ~ NA_character_
)

# larger and more invasive it is the higher number it is. T4 > T1
table(merged_data$T_stage_simple)

fit <- survfit(Surv(OS_time, OS_event) ~ T_stage_simple, data = merged_data)

plot(
  fit,
  col = c("#1b9e77", "#d95f02", "#7570b3", "#e7298a"),
  lwd = 2,
  lty = 1:4,
  xlab = "Time (months)",
  ylab = "Survival Probability",
  main = "Overall Survival by T-Stage (LIHC)"
)

legend(
  "bottomleft",
  legend = levels(factor(merged_data$T_stage_simple)),
  col = c("#1b9e77", "#d95f02", "#7570b3", "#e7298a"),
  lty = 1:4,
  lwd = 2,
  bty = "n"
)

# so this is good. very statiscally significant
#Survival analysis by AJCC T-stage (T1–T4) revealed a strong association between tumor invasiveness and patient outcomes. After extracting the primary tumor stage from the clinical dataset, patients were grouped into T1 (n = 174), T2 (n = 87), T3 (n = 75), and T4 (n = 13). Kaplan–Meier curves showed a clear stepwise decline in survival probability with increasing T-stage, with T4 patients demonstrating the poorest outcomes.

#A log-rank test confirmed that these differences were highly statistically significant (χ² = 35, p = 1×10⁻⁷). This indicates that tumor size and local invasion are strong prognostic factors in LIHC, consistent with known clinical behavior of hepatocellular carcinoma and with trends reported in the TCGA LIHC study. - chat.
```


